File: ios/EventEmitter/EventEmitter.m
//
//  EventEmitter.m
//  CloudpaymentsSdk
//
//  Created by Anton Votinov on 01.02.2022.
//  Copyright © 2022 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>
#import <Foundation/Foundation.h>

@interface RCT_EXTERN_MODULE(EventEmitter, RCTEventEmitter)
  RCT_EXTERN_METHOD(supportedEvents)
@end

File: ios/EventEmitter/EventEmitter.swift
//
//  EventEmitter.swift
//  CloudpaymentsSdk
//
//  Created by Anton Votinov on 01.02.2022.
//  Copyright © 2022 Facebook. All rights reserved.
//

import Foundation

@objc(EventEmitter)
open class EventEmitter: RCTEventEmitter {
  public static var emitter: RCTEventEmitter!
  private let listenerCryptogramCard = "listenerCryptogramCard"
    
  override init() {
    super.init()
    EventEmitter.emitter = self;
  }

  @objc
  open override func supportedEvents() -> [String] {
      [self.listenerCryptogramCard]
  }

  @objc
  public override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
